File: pkg/checkpoint/inmemorycheckpointer.go
/*
Copyright 2022 IBM All Rights Reserved.

SPDX-License-Identifier: Apache-2.0
*/

package checkpoint

type InMemoryCheckpointer struct {
	blockNumber uint64
	transactionID string
}

func (c *InMemoryCheckpointer) CheckpointBlock(blockNumber uint64) {
	c.blockNumber = blockNumber + 1
	c.transactionID = ""
}

func (c *InMemoryCheckpointer) CheckpointTransaction(blockNumber uint64 , transactionID string) {
	c.blockNumber = blockNumber
	c.transactionID = transactionID
}

func (c *InMemoryCheckpointer) BlockNumber() uint64 {
	return c.blockNumber
}

func (c *InMemoryCheckpointer) TransactionID() string {
	return c.transactionID
}

File: java/src/main/java/org/hyperledger/fabric/client/InMemoryCheckpointer.java
/*
 * Copyright 2022 IBM All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package org.hyperledger.fabric.client;

import java.util.Optional;
import java.util.OptionalLong;

/**
 * A non-persistent Checkpointer implementation.
 * It can be used to checkpoint progress after successfully processing events, allowing eventing to be resumed from this point.
 */
public final class InMemoryCheckpointer implements Checkpointer {

    private OptionalLong blockNumber = OptionalLong.empty();
    private String transactionId;

    @Override
    public void checkpointBlock(final long blockNumber) {
        checkpointTransaction(blockNumber + 1, null);
    }

    @Override
    public void checkpointTransaction(final long blockNumber, final String transactionId) {
        this.blockNumber = OptionalLong.of(blockNumber);
        this.transactionId = transactionId;
    }

    @Override
    public void checkpointChaincodeEvent(final ChaincodeEvent event) {
        checkpointTransaction(event.getBlockNumber(), event.getTransactionId());
    }

    @Override
    public OptionalLong getBlockNumber() {
        return blockNumber;
    }

    @Override
    public Optional<String> getTransactionId() {
        return Optional.ofNullable(transactionId);
    }
}
